packages/lodestar/src/network/reqResp.ts

```diff
@@ -80,7 +84,7 @@ export class ReqResp extends (EventEmitter as IReqEventEmitterClass) implements
             pipe(
               stream.source,
               eth2RequestDecode(this.config, this.logger, method, encoding),
-              this.handleRpcRequest(peerId, method),
+              this.handleRpcRequest(peerId, method, encoding),
```

Comment:
Feels like it would be a lot cleaner if you add additional function before request handling that would store encoding instead of passing that troughout stack.
like:
```typescript
pipe(
stream.source,
eth2RequestDecode,
storePeerEncodingPreference,
this.handleRpcRequest
)
```
This pipe will respond with proper encoding anyways, issue is just when we are sending request.